packages/relay/src/lib/eth.ts

```
@@ -499,13 +499,14 @@ export class EthImpl implements Eth {
       // retrieve the contract result details
       await this.mirrorNodeClient.getContractResultsDetails(address, contractResult.results[0].timestamp)
         .then(contractResultDetails => {
+          if(contractResultDetails === null) {
```

we do want to do this null check here because we want to throw the RESOURCE_NOT_FOUND exception if the contract details are not found at all.

---

BTW, now can lose the `?` on line 502 ...

---

yes - nice catch